### ISSUE TITLE
[deeplab]traing on my own dataset, produce nothing.

### ISSUE THREAD

**Issue by aohan237** [tensorflow/models#7486]
read the deeplab docs.

i made my owndataset, by modify the data_generator.py and build_*dataset.py.

then i use xexception_71 as the train_init_ckpt, and train my own model

> --initialize_last_layer=false 

> --fine_tune_batch_norm=false 

when i run vis.py , it detects nothing, why.

i have tried several ways:
1、train my model from scratch. the loss grows bigger and then nan. failed.
2、modify the training_batch and learning_rate related argus, the loss keeps constant
3、i modify the until/train_util.py about scored_label's weights, i make backgroud label weight is 1,and object weight is 20, and then loss becomes good.  but it detection is werid, it really found something,but something is the wrong location and the label is wrong,too. 
at first ,i thouth it may be the training step is too small.  then i trained for 50k steps, the result is still the same.
4、try the local_test.sh on pacal dataset,it produces fine, but the model only trained for 10step, so maybe is the result of the pretrained model.  then i modify my dataset to adapt the test format, the result is the same as above.
5、in order to exclude the reason of the dataset, i trained the ade20k dataset use pacsal pretrained model, the result  is still the same as above.

hope you guys can help me with this, thanks

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
What is the top-level directory of the model you are using
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by aohan237** [comment]
What is the top-level directory of the model you are using: research
Have I written custom code: yes, custom dataset
OS Platform and Distribution: linux,ubuntu 18.04
TensorFlow installed from: pip
TensorFlow version: 1.13.1
Bazel version:None
CUDA/cuDNN version:1.7
GPU model and memory: yes, 11g*4
Exact command to reproduce: doc model

**Comment by lattard** [comment]
hi, is there any update on this please ? I'm having the same issue.

**Comment by aohan237** [comment]
@lattard  to use deeplab model,you have to read the source code , otherwise you wont get the right result only through the doc.

here is some instructions for new comers:

1.  the object label in image, must start from 0, and then as a sequence.  cause, the source code ,use tensorflow one hot  to transfer these labels.  if you dont know one hot,may be you try to learn it.

2.  retraining Batch norm will be very important for your custom dataset.  as in deeplab, maybe  --fine_tune_batch_norm=True. but first you may need a good machine.
   batch_size need a bit large for fine tune batch_norm.

3.  xception model is a little big for fresh man.  you may need mobilenet as a backbone. and atrous_rates only make sense if you use xception backbone.

4. if you still get nothing, you can  modify utils/train_utils.py to manually modify the weights corresponding to the labels.

after these steps,you maybe get what you want

4.
